cou/commands.py

```diff
@@ -0,0 +1,343 @@
+        "data-plane",
+        description="Show the steps for upgrading the data-plane components. This is possible "
+        "only if control-plane has been fully upgrade. Otherwise an error will be thrown.",
+        help="Show the steps for upgrading the data-plane components. This is possible "
+        "only if control-plane has been fully upgrade. Otherwise an error will be thrown.",
+        usage="cou plan data-plane [options]",
+        parents=[subcommand_common_opts_parser, dp_parser],
+        formatter_class=CapitalizeHelpFormatter,
+    )
+
+
+def create_run_subparser(
+    subparsers: argparse._SubParsersAction,
+    subcommand_common_opts_parser: argparse.ArgumentParser,
+    dp_parser: argparse.ArgumentParser,
+) -> None:
+    """Create and configure 'run' subcommand parser.
+
+    :param subparsers: subparsers that plan subparser belongs to
+    :type subparsers: argparse.ArgumentParser
+    :param subcommand_common_opts_parser parser groups options commonly shared by subcommands
+    :type subcommand_common_opts_parser: argparse.ArgumentParser
+    :param dp_parser: a parser groups options specific to data-plane
+    :type dp_parser: argparse.ArgumentParser
+    """
+    # Arg parser for "cou run" sub-command and set up common options
+    run_args_parser = argparse.ArgumentParser(add_help=False)
+    run_args_parser.add_argument(
+        "--interactive",
+        help="Run upgrade with prompt.",
+        action=argparse.BooleanOptionalAction,
+        default=True,
+    )
+    run_parser = subparsers.add_parser(
+        "run",
+        description="Run the cloud upgrade. If upgrade-group is unspecified, perform "
+        "upgrade for the entire cloud",
+        help="Run the cloud upgrade.",
+        usage="cou run [options]",
+        parents=[subcommand_common_opts_parser, run_args_parser],
+        formatter_class=CapitalizeHelpFormatter,
+    )
+
+    # Create control-plane and data-plane child commands to run partial upgrades
+    run_subparser = run_parser.add_subparsers(
+        title="Upgrade group",
+        dest="upgrade-group",
+        help="For more information about an upgrade group, run 'cou run <upgrade-group> -h'.",
+    )
+    run_subparser.add_parser(
+        "control-plane",
+        description="Run upgrade for the control-plane components.",
+        help="Run upgrade for the control-plane components.",
+        usage="cou plan control-plane [options]",
+        parents=[subcommand_common_opts_parser, run_args_parser],
+        formatter_class=CapitalizeHelpFormatter,
+    )
+    run_subparser.add_parser(
+        "data-plane",
+        description="Run upgrade for the data-plane components. This is possible only if "
+        "control-plane has been fully upgrade. Otherwise an error will be thrown.",
+        help="Run upgrade for the data-plane components. This is possible only if "
+        "control-plane has been fully upgrade. Otherwise an error will be thrown.",
```

Comment:
```suggestion
        description="Run upgrade for the data-plane components. This is possible only if "
        "control-plane has been fully upgraded. Otherwise an error will be thrown.",
        help="Run upgrade for the data-plane components. This is possible only if "
        "control-plane has been fully upgraded. Otherwise an error will be thrown.",
```